### ISSUE TITLE
Register UDF cause node crash

### ISSUE THREAD

**Issue by submarineblue** [aerospike/aerospike-server#174]
l was trying to register my udfs using aql. Several beginning registers succeeded and suddenly one register failed with info "`Error: (-1) Bad file descriptor" and the following retry also failed with info "Error: (-1) Socket write error: 111`". After l checked the asd instances, lt was found that all the instances crashed due to the udf register.

My version is Aerospike community Edition build 3.9.0, and 4 nodes are deployed. The crash stack is:
```
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: found 11 frames
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 0: /home/search/aerospike-server/bin/asd(as_sig_handle_segv+0x3a) [0x487780]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 1: /lib64/libc.so.6 [0x39894302d0]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 2: /home/search/aerospike-server/bin/asd(lua_pushcclosure+0x9) [0x544399]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 3: /home/search/aerospike-server/bin/asd(luaL_openlibs+0x1a) [0x552b4a]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 4: /home/search/aerospike-server/bin/asd [0x529895]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 5: /home/search/aerospike-server/bin/asd(cache_init+0x195) [0x52b956]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 6: /home/search/aerospike-server/bin/asd [0x52c16c]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 7: /home/search/aerospike-server/bin/asd(udf_cask_smd_accept_fn+0x2ef) [0x4c23cd]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 8: /home/search/aerospike-server/bin/asd(as_smd_thr+0x12b7) [0x49a7ee]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 9: /lib64/libpthread.so.0 [0x3989c064a7]
Mar 02 2017 02:19:47 GMT: WARNING (as): (signal.c:185) stacktrace: frame 10: /lib64/libc.so.6(clone+0x6d) [0x39894d3c2d]
```

**Comment by bbulkow** [comment]
You do know your version is about 6 months out of date, right? We're up to 3.11.1 now.

**Comment by submarineblue** [comment]
Do you mean that l should upgrade to the latested version? 'cause it would cost too much to upgrade program in production system.

**Comment by kportertx** [comment]
I found a similar item addressed in [3.10.0.3](http://www.aerospike.com/download/server/notes.html#3.10.0.3), though it said on startup.

**Comment by kportertx** [comment]
Confirmed that AER-5258 not only happened on startup but also registration. I suggest upgrading to resolve this issue.
